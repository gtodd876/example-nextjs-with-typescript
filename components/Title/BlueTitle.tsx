import styled from 'styled-components';
import { Title } from './Title';

export const BlueTitle = styled(Title)`
  color: blue;
`;
