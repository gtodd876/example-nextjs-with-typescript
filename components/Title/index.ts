export * from './Title';
export * from './BlueTitle';
