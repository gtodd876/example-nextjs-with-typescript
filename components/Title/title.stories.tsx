import React from 'react';
import { storiesOf } from '@storybook/react';
import { Title } from './Title';
import { BlueTitle } from './BlueTitle';

const stories = storiesOf('Button', module);

stories.add('Default', () => <Title>Title</Title>);
stories.add('Blue Title', () => <BlueTitle>Title</BlueTitle>);
