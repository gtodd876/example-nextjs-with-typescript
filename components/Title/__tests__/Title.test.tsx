import React from 'react';
import { render } from '@testing-library/react';
import { Title } from '../Title';

describe('Title', () => {
  test('Styled Title component', () => {
    const { container } = render(<Title>Test</Title>);
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('font-size', '50px');
  });
});
