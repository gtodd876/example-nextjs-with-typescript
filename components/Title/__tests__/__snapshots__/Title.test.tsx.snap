// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Styled Title component 1`] = `
.c0 {
  font-size: 50px;
}

<div>
  <h1
    class="c0"
  >
    Test
  </h1>
</div>
`;
